service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/EhrResendControllerTest.java

```diff
@@ -0,0 +1,269 @@
+                                                           .conversationClosed(FIVE_DAYS_AGO)
+                                                           .errors(List.of(
+                                                               EhrExtractStatus.EhrReceivedAcknowledgement.ErrorDetails.builder()
+                                                                   .code(INCUMBENT_NACK_CODE)
+                                                                   .display(INCUMBENT_NACK_DISPLAY)
+                                                                   .build()))
+                                                           .messageRef(ehrMessageRef)
+                                                           .received(FIVE_DAYS_AGO)
+                                                           .rootId(generateRandomUppercaseUUID())
+                                                           .build());
+        ehrExtractStatus.setEhrRequest(EhrExtractStatus.EhrRequest.builder().nhsNumber(NHS_NUMBER).build());
+
+        doReturn(Optional.of(ehrExtractStatus)).when(ehrExtractStatusRepository).findByConversationId(CONVERSATION_ID);
+
+        var response = ehrResendController.scheduleEhrExtractResend(CONVERSATION_ID);
+
+        assertEquals(HttpStatus.ACCEPTED, response.getStatusCode());
+        assertNull(response.getBody());
+    }
+
+    @Test
+    void When_AnEhrExtractHasNotFailed_Expect_RespondsWith403() throws JsonProcessingException {
+
+        var details = new CodeableConcept();
+        var codeableConceptCoding = new Coding();
+        codeableConceptCoding.setSystem(GPCONNECT_ERROR_OR_WARNING_CODE);
+        codeableConceptCoding.setCode(INTERNAL_SERVER_ERROR);
+        details.setCoding(List.of(codeableConceptCoding));
+
+        String ehrMessageRef = generateRandomUppercaseUUID();
+        var ehrExtractStatus = new EhrExtractStatus();
+
+        ehrExtractStatus.setConversationId(CONVERSATION_ID);
+        ehrExtractStatus.setEhrReceivedAcknowledgement(EhrExtractStatus.EhrReceivedAcknowledgement.builder()
+                                                           .conversationClosed(FIVE_DAYS_AGO)
+                                                           .errors(List.of())
+                                                           .messageRef(ehrMessageRef)
+                                                           .received(FIVE_DAYS_AGO)
+                                                           .rootId(generateRandomUppercaseUUID())
+                                                           .build());
+        ehrExtractStatus.setEhrRequest(EhrExtractStatus.EhrRequest.builder().nhsNumber(NHS_NUMBER).build());
+
+        doReturn(Optional.of(ehrExtractStatus)).when(ehrExtractStatusRepository).findByConversationId(CONVERSATION_ID);
+
+        var response = ehrResendController.scheduleEhrExtractResend(CONVERSATION_ID);
+
+        JsonNode rootNode = objectMapper.readTree(response.getBody());
```

Comment:
Really not sure what is going on with this?  Why are we using a JsonTree and navigating it, rather than just reading the value into an object? The rootNode makes it a whole less readable

```java
var operationOutcome = objectMapper.readValue(response.getBody(), OperationOutcome.class);
        var coding = operationOutcome.getIssue().getFirst().getDetails().getCoding().getFirst()
```